### PR TITLE
[FIX][IMP] mrp,stock: some usability changes

### PR DESCRIPTION
**Pull request by svs-odoo**
A few changes, fixes or improvements for `mrp`, `purchase_stock` and `stock`.

See commit's message for more information.

task-2328830

### CODE REVIEW

File: addons/mrp/models/mrp_production.py [hunk 1]
```diff
@@ -741,6 +741,13 @@ def write(self, vals):
 
     @api.model
     def create(self, values):
+        # Remove from `move_finished_ids` the by-product moves and then move `move_byproduct_ids`
+        # into `move_finished_ids` to avoid duplicate and inconsistency.
+        if values.get('move_finished_ids', False):
```
Comment (Whenrow) [inline addons/mrp/models/mrp_production.py:746]: I think i will be cleaner to do the same process than the move_without_package magic 
https://github.com/odoo/odoo/blob/5e5de2b2b64fd8cb259e72fc1a97dc754da871ad/addons/stock/models/stock_picking.py#L778-L805

Re-computing the byproduct_move_ids each time it's updated

File: addons/stock/models/stock_picking.py [hunk 2]
```diff
@@ -324,6 +324,7 @@ class Picking(models.Model):
         readonly=True)
     picking_type_entire_packs = fields.Boolean(related='picking_type_id.show_entire_packs',
         readonly=True)
+    hide_picking_type = fields.Boolean(compute='_compute_hide_pickign_type', store=False)
```
Comment (Whenrow) [inline addons/stock/models/stock_picking.py:327]: store=False not needed

File: addons/mrp/views/mrp_bom_views.xml [hunk 3]
```diff
@@ -90,7 +90,7 @@
                                     <field name="parent_product_tmpl_id" invisible="1" />
                                     <field name="possible_bom_product_template_attribute_value_ids" invisible="1"/>
                                     <field name="product_uom_id" options="{'no_open':True,'no_create':True}" groups="uom.group_uom"/>
-                                    <field name="bom_product_template_attribute_value_ids" widget="many2many_tags" options="{'no_create': True}" attrs="{'column_invisible': [('parent.product_id', '!=', False)]}" groups="product.group_product_variant"/>
+                                    <field name="bom_product_template_attribute_value_ids" optional="hidden" widget="many2many_tags" options="{'no_create': True}" attrs="{'column_invisible': [('parent.product_id', '!=', False)]}" groups="product.group_product_variant"/>
```
Comment (Whenrow) [inline addons/mrp/views/mrp_bom_views.xml:93]: hide and hidden work the same ? pfff misère

Comment (svs-odoo) [reply addons/mrp/views/mrp_bom_views.xml:93]: Yep, but as `hide` is more used, I will use this one instead :+1:

**Comment by Whenrow** [comment]
robodoo r+ rebase-ff

**Pull request closed without merging**